proto-build/src/main.rs

```diff
@@ -434,8 +440,13 @@ fn copy_and_patch(src: impl AsRef<Path>, dest: impl AsRef<Path>) -> io::Result<(
         .replace_all(&contents, "tendermint_proto");
 
     // Patch each service definition with a feature attribute
-    let patched_contents =
-        contents.replace(TONIC_CLIENT_ATTRIBUTE, &GRPC_CLIENT_ATTRIBUTES.join("\n"));
+    let contents = contents.replace(TONIC_CLIENT_ATTRIBUTE, &GRPC_CLIENT_ATTRIBUTES.join("\n"));
 
-    fs::write(dest, patched_contents)
+    // Feature gate gRPC client impl blocks which rely on `tonic::transport`
+    let contents = Regex::new(TONIC_CLIENT_IMPL)
+        .unwrap()
+        .replace_all(&contents, TONIC_CLIENT_REPLACEMENT);
+
+    // Write output to destination
+    fs::write(dest, &*contents)
```

Comment:
It might be worth refactoring how these transformations are applied to the generated protos, making it into an folding iterator over `Regex` and replacement string tuples that applies successive transformations